src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -127,32 +127,56 @@ public static function register($mode = false)
-                        echo "\n", $colorize(sprintf('%s deprecation notices (%d)', ucfirst($group), $deprecations[$group.'Count']), 'legacy' !== $group), "\n";
+                $displayDeprecations = function ($deprecations) use ($colorize, $cmp) {
+                    foreach (array('unsilenced', 'remaining', 'legacy', 'other') as $group) {
+                        if ($deprecations[$group.'Count']) {
+                            echo "\n", $colorize(sprintf('%s deprecation notices (%d)', ucfirst($group), $deprecations[$group.'Count']), 'legacy' !== $group), "\n";
 
-                        uasort($deprecations[$group], $cmp);
+                            uasort($deprecations[$group], $cmp);
 
-                        foreach ($deprecations[$group] as $msg => $notices) {
-                            echo "\n  ", $notices['count'], 'x: ', $msg, "\n";
+                            foreach ($deprecations[$group] as $msg => $notices) {
+                                echo "\n  ", $notices['count'], 'x: ', $msg, "\n";
 
-                            arsort($notices);
+                                arsort($notices);
 
-                            foreach ($notices as $method => $count) {
-                                if ('count' !== $method) {
-                                    echo '    ', $count, 'x in ', preg_replace('/(.*)\\\\(.*?::.*?)$/', '$2 from $1', $method), "\n";
+                                foreach ($notices as $method => $count) {
+                                    if ('count' !== $method) {
+                                        echo '    ', $count, 'x in ', preg_replace('/(.*)\\\\(.*?::.*?)$/', '$2 from $1', $method), "\n";
+                                    }
                                 }
                             }
                         }
                     }
-                }
-                if (!empty($notices)) {
-                    echo "\n";
-                }
+                    if (!empty($notices)) {
+                        echo "\n";
+                    }
+                };
+
+                $displayDeprecations($deprecations);
 
-                if ('weak' !== $mode && ($deprecations['unsilenced'] || $deprecations['remaining'] || $deprecations['other'])) {
-                    exit(1);
+                // store failing status
+                $isFailing = 'weak' !== $mode && ($deprecations['unsilenced'] || $deprecations['remaining'] || $deprecations['other']);
+
+                // reset deprecations array
+                foreach ($deprecations as $key => $arrayOrInt) {
+                    if (is_int($arrayOrInt)) {
+                        $deprecations[$key] = 0;
+                    } else {
+                        $deprecations[$key] = array();
+                    }
                 }
+
+                register_shutdown_function(function () use (&$deprecations, $isFailing, $displayDeprecations) {
+                    if (array_sum(array_filter($deprecations, function ($key) {
```

Comment:
This calls for a data structure refactoring

Reply:
let's do a code-style fix for now :) the "if" should be on one line

Reply:
"fixed" :P